index.js

```diff
@@ -487,6 +489,21 @@ function flameGraph (opts) {
     }
   }
 
+  function getNodeRect(node){
```

Comment:
A couple of thoughts on this:

1. Since this returns the node rect co-ordinates relative to the page / scroll position, not the raw canvas co-ordinates, can we have a  name that reflects that, like `getNodePositionOnPage`, and maybe also a comment saying which element the returned co-ordinates are relative to (presumably, it's the div wrapper the canvas is inside?)
2. It seems a little Flame-specific. For example, _we_ don't need to worry about horizontal scroll position, or CSS padding between wrapper and canvas, because Flame (and 0x) scales X to fit responsively and doesn't pad - but _d3-fg_ doesn't know that. This logic also presumably depends on the target element being `position: relative;` in the host application's UI? Maybe it'd make more sense to pass an options object including `h` and `c` to the event handlers, then move this logic to Flame? (this will also make it easier to work on in future :-) )

Reply:
The `rect` co-ordinates are relative to the **canvas**, not the page. They return the node distance from the canvas top/left corner and it's size.
The `wrapper.node().scrollTop` was actually not needed and I removed it.
I don't see how this is Flame-specific code, it just triggers a couple of events passing values calculated in the canvas context.
I just created a new branch with these tweaks and I created a PR from there so please [review and comment here](https://github.com/davidmarkclements/d3-fg/pull/15)

Reply:
yea, I think emitting the rect (relative to canvas) like this does now is the best approach—most of the calculations here _need_ to happen in d3-fg because the outside world (Flame) doesn't know about the zoomTransform. The zoomTransform needs to be applied after the scaleToWidth stuff, so we can't just move that out either; the zoomTransform may contain both scaling and translation, so the order is important.